static/app/views/alerts/rules/issue/index.tsx

```diff
@@ -694,9 +694,12 @@ class IssueRuleEditor extends AsyncView<Props, State> {
     this.setState(prevState => {
       const clonedState = cloneDeep(prevState);
 
-      const newTypeList = prevState.rule ? prevState.rule[type] : [];
-      if (prevState.rule) {
-        newTypeList.splice(idx, 1);
```

Comment:
splice is a bit easier to tell what the function is trying to do, can we create a new array instead and keep splice.
```
newTypeList.splice(idx, 1)
newTypeList = [...newTypeList]
```